service/handlers_test.go

```diff
@@ -432,121 +368,69 @@ func TestDeleteTarget(t *testing.T) {
-			want:    http.StatusBadRequest,
-			asAdmin: true,
-			method:  "POST",
-			url:     "/workflows",
-		},
-		{
-			name:    "project name must be present",
-			req:     loadCreateWorkflowRequest(t, "TestCreateWorkflow/project_name_must_be_present.json"),
-			want:    http.StatusBadRequest,
-			asAdmin: true,
-			method:  "POST",
-			url:     "/workflows",
-		},
-		{
-			name:    "target name must be present",
-			req:     loadCreateWorkflowRequest(t, "TestCreateWorkflow/target_name_must_be_present.json"),
-			want:    http.StatusBadRequest,
-			asAdmin: true,
-			method:  "POST",
-			url:     "/workflows",
+			name:     "can create workflows",
+			req:      loadJSON(t, "TestCreateWorkflow/can_create_workflow_request.json"),
+			want:     http.StatusOK,
+			respFile: "TestCreateWorkflow/can_create_workflow_response.json",
+			method:   "POST",
+			url:      "/workflows",
 		},
+		// We test this specific validation as it's server side only.
 		{
-			name:    "type must be present",
-			req:     loadCreateWorkflowRequest(t, "TestCreateWorkflow/type_must_be_present.json"),
-			want:    http.StatusBadRequest,
-			asAdmin: true,
-			method:  "POST",
-			url:     "/workflows",
+			name:     "framework must be valid",
+			req:      loadJSON(t, "TestCreateWorkflow/framework_must_be_valid_request.json"),
+			want:     http.StatusBadRequest,
+			respFile: "TestCreateWorkflow/framework_must_be_valid_response.json",
+			method:   "POST",
+			url:      "/workflows",
 		},
+		// We test this specific validation as it's server side only.
 		{
-			name:    "type must be valid",
-			req:     loadCreateWorkflowRequest(t, "TestCreateWorkflow/type_must_be_valid.json"),
-			want:    http.StatusBadRequest,
-			asAdmin: true,
-			method:  "POST",
-			url:     "/workflows",
+			name:     "type must be valid",
+			respFile: "TestCreateWorkflow/type_must_be_valid_response.json",
+			req:      loadJSON(t, "TestCreateWorkflow/type_must_be_valid_request.json"),
+			want:     http.StatusBadRequest,
+			method:   "POST",
+			url:      "/workflows",
 		},
 		{
-			name:    "project must exist",
-			req:     loadCreateWorkflowRequest(t, "TestCreateWorkflow/project_must_exist.json"),
-			want:    http.StatusBadRequest,
-			asAdmin: true,
-			method:  "POST",
-			url:     "/workflows",
+			name:   "project must exist",
+			req:    loadJSON(t, "TestCreateWorkflow/project_must_exist.json"),
+			want:   http.StatusBadRequest,
+			method: "POST",
+			url:    "/workflows",
 		},
 		{
-			name:    "target must exist",
-			req:     loadCreateWorkflowRequest(t, "TestCreateWorkflow/target_must_exist.json"),
-			want:    http.StatusBadRequest,
-			asAdmin: true,
-			method:  "POST",
-			url:     "/workflows",
+			name:   "target must exist",
+			req:    loadJSON(t, "TestCreateWorkflow/target_must_exist.json"),
+			want:   http.StatusBadRequest,
+			method: "POST",
+			url:    "/workflows",
 		},
+		// TODO with admin credentials should fail
```

Comment:
I found a number of auth related items. Those should be addressed in the future auth refactor.